src/plugins/data/public/query/query_string/dataset_service/types.ts

```diff
@@ -106,4 +106,8 @@ export interface DatasetTypeConfig {
    * Service used for indexedViews related operations
    */
   indexedViewsService?: DatasetIndexedViewsService;
+  /**
+   * Returns the default query that is added to the query editor when a dataset is selected.
+   */
+  getDefaultQueryString?: (dataset: Dataset, language: string) => string | void;
```

Comment:
the language service passes in a `Query` object since it already has dataset and language. 

Could we provide this as well? If functionality it is similar to the language config `getQueryString` wouldn't it be better to call it `getQueryString`. And I don't think we want to return type of `string | void` if people decide to implement and return `void` then this function name is confusing.

Reply:
Let's go with `getInitialQueryString`